src/dune_rules/compile_time.mli

```diff
@@ -22,3 +22,4 @@ val pkg_build_progress : Config.Toggle.t Config.t
 val lock_dev_tools : Config.Toggle.t Config.t
 
 val bin_dev_tools : Config.Toggle.t Config.t
+val portable_lock_dir : Config.Toggle.t Config.t
```

Comment:
I think this flag isn't exactly what we're looking for. The feature itself should always be enabled, the only thing that should be configurable at build time is the default configuration we're solving for. Classically, this should be just be the "current" configuration. With this feature, it might be something like `:current $linux $win $mac` for example.

Reply:
Indeed, the only difference in behaviour when this flag is enabled is that the solver is run for a list of platforms rather than just the current platform. What are you suggesting be changed here?

Reply:
The lockfiles also gain some extra fields and some fields can now contain conditional values that depend on the platform. Currently I've kept the lockfile parser/enocder backwards compatible so I don't have to modify all our tests to use the new syntax just yet.